src/main/java/com/oneandone/snmpman/configuration/modifier/CommunityIndexCounter32Modifier.java

```diff
@@ -0,0 +1,85 @@
+package com.oneandone.snmpman.configuration.modifier;
+
+import com.google.common.base.Optional;
+import com.oneandone.snmpman.configuration.type.ModifierProperties;
+import lombok.Getter;
+import org.snmp4j.smi.Counter32;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.OctetString;
+import org.snmp4j.smi.Variable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * This modifier implementation modifies {@link org.snmp4j.smi.Counter32} variables depending on their community context.
+ */
+public class CommunityIndexCounter32Modifier implements CommunityContextModifier<Counter32> {
+    /**
+     * Mapping of SNMP community context to SNMP OID and result.
+     */
+    @Getter private Map<Long, Map<OID, Long>> communityContextMapping = new HashMap<>();
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void init(final ModifierProperties properties) {
+        communityContextMapping = new HashMap<>();
+        properties.entrySet().stream().filter(property -> getUnsignedLong(property.getKey()) != -1L).forEach(property -> {
+            final Long context = getUnsignedLong(property.getKey());
+            final Object propertyValue = property.getValue();
+            final Map<Object, Object> oidValueMapping = (Map<Object, Object>) propertyValue;
+            final Map<OID, Long> result = new HashMap<>();
+            oidValueMapping.entrySet().stream().filter(ovm -> getUnsignedLong(ovm.getValue()) != -1L).forEach(ovm -> {
+                final OID oid = new OID((String) ovm.getKey());
+                final Long value = getUnsignedLong(ovm.getValue());
+                result.put(oid, value);
+            });
+            communityContextMapping.put(context, result);
+        });
+    }
+
+    private Long getUnsignedLong(final Object input) {
+        try {
+            if (!Optional.fromNullable(input).isPresent()) {
+                // not present
+                return -1L;
+            }
+            final String value = String.valueOf(input);
+            return Long.parseLong(value);
+        } catch (NumberFormatException e) {
+            // s is not numeric
+            return -11L;
+        }
+    }
+
+    @Override
+    public final Counter32 modify(final Counter32 variable) {
+        if (variable == null) {
+            return new Counter32(0);
+        }
+        return variable;
+    }
+
+    @Override
+    public Map<OID, Variable> getVariableBindings(final OctetString context, final OID queryOID) {
+        if (queryOID != null && context != null && context.getValue().length != 0) {
+            if (!queryOID.toString().isEmpty() && !context.toString().isEmpty()) {
+                if (communityContextMapping.containsKey(Long.parseLong(context.toString())) &&
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/1and1_NDev/snmpman/file/2405103234/issues/source?bid=3233453&fileBranchId=3233458#l68)